File: heeyaa/Learn/Array101/max_consecutive_ones.py
from typing import List

# ========Answer Part ========
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count, result = 0, 0
        
        for num in nums:
            if num == 1:
                count += 1
                result = max(count, result)
            else:
                count = 0
                
        return result
# =============================


if __name__ == '__main__':
    nums = [1,1,0,1,1,1]
    obj = Solution()
    print(obj.findMaxConsecutiveOnes(nums))

File: sun0/Learn/Array101/max_consecutive_ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0;
        int maxCount = 0;
        for (int a : nums) {
            if (a == 1) {
                count ++;
                // max함수로 최대값 비교
                maxCount = max(maxCount, count);
            }
            else {
                count = 0;
            }
        }
        return maxCount;
    }
};

// runtime:47ms, memoryusage: 36.2mb